### PR TITLE
User Licensing: Add post-checkout UI for auto-activation flow

### PR DESCRIPTION
**Pull request by elliottprogrammer**
#### Changes proposed in this Pull Request

- This PR adds the post-checkout thank you page for users that have at least one jetpack site connected.
- Users select which site they want to activate their product on from a drop-down list of all their sites.
- Or they can choose, "I don't see my site. Let me configure it manually", in which they will be redirected to the manual activation flow.
- If the user does not have any connected Jetpack sites, they will skip this UI and be redirected straight to the "manual"-activation flow.

Related to 1201096622142517-as-1201096622142518

#### Testing instructions

- Create a JN site & connect to your account.
- Checkout this PR and `yarn-start`.
- Go to: http://localhost:3000/checkout/jetpack/jetpack_scan
- Complete purchase.  Verify you are redirected to: `/checkout/jetpack/thank-you/licensing-auto-activate/:product` and you can see the the new post-checkout UI (See screenshot below).
- In the dropdown select box, select, "I don't see my site. Let me configure it manually" and hit Continue.
- Verify you are taken to the manual license-activation UI flow (This UI will need more work in future PR's)
- Click the browsers "Back" button to navigate back to the auto-activation flow.
- This time in the dropdown select box, select your JN site that you connected earlier, and hit continue.
- Verify the Jetpack Scan subscription was successfully transferred to your JN site and you are redirected to the "<product> has been activated and is ready to go!". page.

#### Screenshots:

Screenshot on 2021-11-01 at 16-48-55

Markup 2021-11-02 at 12 25 18

### CODE REVIEW

File: client/my-sites/checkout/checkout-thank-you/licensing-thank-you-auto-activation.tsx [hunk 1]
```diff
@@ -0,0 +1,236 @@
...
+							{ translate( 'Select the site you want %(productName)s on:', {
+								args: {
+									productName,
+								},
+							} ) }
+						</p>
+					) }
+					<SelectDropdown
+						className="licensing-thank-you-auto-activation__select"
+						selectedText={ selectedItem ? selectedItem.label : translate( 'Select…' ) }
```
Comment (a8ci18n) [inline client/my-sites/checkout/checkout-thank-you/licensing-thank-you-auto-activation.tsx:189]: :information_source: String reuse speeds up translation and improves consistency. The following string might make a good alternative and has already been translated **125** times:
`translate( '- Select -' )` *ES Score: 6*

File: client/my-sites/checkout/checkout-thank-you/licensing-thank-you-manual-activation.tsx [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+			<Card className="licensing-thank-you-manual-activation__card">
+				<div className="licensing-thank-you-manual-activation__card-main">
+					<JetpackLogo size={ 45 } />
+					{ hasProductInfo && <QueryProducts type="jetpack" /> }
+					<h1 className="licensing-thank-you-manual-activation__main-message">
+						{ translate( 'Thank you for your purchase!' ) }{ ' ' }
+						{ String.fromCodePoint( 0x1f389 ) /* Celebration emoji 🎉 */ }
+					</h1>
+					<p className="licensing-thank-you-manual-activation__product-info">
+						{ translate( 'Learn how to get started with Jetpack.' ) }
```
Comment (a8ci18n) [inline client/my-sites/checkout/checkout-thank-you/licensing-thank-you-manual-activation.tsx:59]: :information_source: String reuse speeds up translation and improves consistency. The following string might make a good alternative and has already been translated **19** times:
`translate( 'How to get started with Jetpack' )` *ES Score: 7*
See 1 additional suggestions in the [PR translation status page](https://translate.wordpress.com/localci/status/automattic/wp-calypso/add/licensing-checkout-thank-you-1)

File: client/my-sites/checkout/checkout-thank-you/licensing-thank-you-manual-activation.tsx [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+						<br />
+						{ translate( "We've also sent you an email with these instructions." ) }
+					</p>
+					<Button
+						className="licensing-thank-you-manual-activation__button"
+						primary
+						disabled={ false }
+					>
+						{ translate( 'Continue ' ) }
```
Comment (a8ci18n) [inline client/my-sites/checkout/checkout-thank-you/licensing-thank-you-manual-activation.tsx:68]: :information_source: String reuse speeds up translation and improves consistency. The following string might make a good alternative and has already been translated **158** times:
`translate( 'Continue' )` *ES Score: 9*

File: client/my-sites/checkout/checkout-thank-you/style.scss [hunk 4]
```diff
@@ -925,18 +934,91 @@
...
 		}
+
+		.licensing-thank-you-auto-activation__card-footer,
+		.licensing-thank-you-manual-activation__card-footer {
+			position: relative;
+			flex: 1 1 456px;
+			display: flex;
+			align-items: center;
+			justify-content: center;
+			background-image: url( '/calypso/images/jetpack/jetpack-licensing-checkout-bg.svg' );
```
Comment (elliottprogrammer) [inline client/my-sites/checkout/checkout-thank-you/style.scss]: For some reason this svg is not rendering properly.. (it appears as a very light-gray background). If I replace this `background-image` declaration with a different svg file, that svg file renders fine, so I believe the problem is this `jetpack-licensing-checkout-bg.svg` file itself.  I exported it from Figma..  Pinging @ederrengifo here for any ideas about what could be wrong with this svg file, or any ideas how to get the svg to display correctly. 🙂

**Comment by rcanepa** [comment]
Some notes after a preliminary review.

> In the dropdown select box, select, "I don't see my site. Let me configure it manually" and hit Continue.

I can't get to the last element on the dropdown (`I don't see my site. Let me configure it manually`) because I have too many sites, and for some reason, the list doesn't support scrolling.

After manually deleting some elements from the list, I could see the last one. I clicked it and worked as expected.

> Click the browsers "Back" button to navigate back to the auto-activation flow.

This didn't work as expected, and I don't know why. I tried to two times and in each case, I was redirected back to the checkout page.

> This time in the dropdown select box, select your JN site that you connected earlier, and hit continue.
Verify the Jetpack Scan subscription was successfully transferred to your JN site and you are redirected to the " has been activated and is ready to go!". page.

This worked as expected!

File: client/my-sites/checkout/checkout-thank-you/style.scss (hunk 4 above)
Comment (rcanepa) [reply client/my-sites/checkout/checkout-thank-you/style.scss]: I see a 404 in the network panel.
image

Comment (rcanepa) [reply client/my-sites/checkout/checkout-thank-you/style.scss]: Maybe there is a better location for assets like this one?

File: client/my-sites/checkout/composite-checkout/hooks/use-get-thank-you-url/get-thank-you-page-url.ts [hunk 5]
```diff
@@ -149,9 +150,9 @@ export default function getThankYouPageUrl( {
 		// extract a product from the cart, in siteless checkout there should only be one
 		const productSlug = cart?.products[ 0 ]?.product_slug;
 
-		const thankYouUrlSiteLess = `/checkout/jetpack/thank-you/no-site/${
-			productSlug ?? 'no_product'
-		}`;
+		const thankYouUrl = isEnabled( 'jetpack/user-licensing' )
+			? `/checkout/jetpack/thank-you/licensing-auto-activate/${ productSlug ?? 'no_product' }`
```
Comment (sirbrillig) [inline client/my-sites/checkout/composite-checkout/hooks/use-get-thank-you-url/get-thank-you-page-url.ts:154]: Could you make sure we add a test for this new URL to `client/my-sites/checkout/composite-checkout/test/composite-checkout-thank-you.js`?

File: client/my-sites/checkout/checkout-thank-you/style.scss (hunk 4 above)
Comment (rcanepa) [reply client/my-sites/checkout/checkout-thank-you/style.scss]: It looks like that after the app is built, the path to the SVG changes. 

I found a workaround that might explain what's going on. I opened the `client/my-sites/checkout/checkout-thank-you/licensing-thank-you-auto-activation.tsx` file, and added this import `import JetpackLicensingBackground from 'calypso/assets/images/jetpack/jetpack-licensing-checkout-bg.svg';` statement.

Then, I replaced line 220 with:
```
				<div className="licensing-thank-you-auto-activation__card-footer" style={ { backgroundImage: `url(${JetpackLicensingBackground})` } }>
```

If you `console.log(JetpackLicensingBackground)` you will see that the path to the SVG is one created dynamically.

Comment (elliottprogrammer) [reply client/my-sites/checkout/checkout-thank-you/style.scss]: Yes this works! Thanks Renzo! 😁

**Comment by rcanepa** [comment]
Thank you for adding a test, Bryan. The background image looks fine now 🎉

**Comment by rcanepa** [comment]
It looks like the height of the dropdown is being determined by the height of `#primary`. If I decrease the window width so that both columns are stacked one on top of each other, then I can see the whole list (this has the effect of increasing the height of `#primary`).

**Comment by elliottprogrammer** [comment]
> It looks like the height of the dropdown is being determined by the height of #primary. If I decrease the window width so that both columns are stacked one on top of each other, then I can see the whole list (this has the effect of increasing the height of #primary).

@rcanepa ,Yes thank you for the additional info. 👍   I changed the primary container to allow overflow (`overflow: visible`), so now a dropdown containing many many sites will extend as far down the page as needed.

Then I also limit the height of the dropdown and put a scrollbar on it, however I'm not sure if scrollbar change is what we want..  What is your opinion?  Dropdown with scrollbar?  Or allow the dropdown to extend down as far as it needs?  (If the list was soo long that the dropdown extends beyond the bottom of the page (beyond the fold), then the entire document would scroll down.)  Let me know your thoughts. 🙂

**Comment by rcanepa** [comment]
> > It looks like the height of the dropdown is being determined by the height of #primary. If I decrease the window width so that both columns are stacked one on top of each other, then I can see the whole list (this has the effect of increasing the height of #primary).
> 
> @rcanepa ,Yes thank you for the additional info. 👍 I changed the primary container to allow overflow (`overflow: visible`), so now a dropdown containing many many sites will extend as far down the page as needed.
> 
> Then I also limit the height of the dropdown and put a scrollbar on it, however I'm not sure if scrollbar change is what we want.. What is your opinion? Dropdown with scrollbar? Or allow the dropdown to extend down as far as it needs? (If the list was soo long that the dropdown extends beyond the bottom of the page (beyond the fold), then the entire document would scroll down.) Let me know your thoughts. 🙂

I think the scrollbar is fine. As you said, the list of sites could be potentially very large so having a scrollbar makes sense to me.

Now, everything looks fine in my opinion. I only caught a very minor issue on mobile which you can see in the following recording. 

https://user-images.githubusercontent.com/3418513/139880542-849e7192-7a4d-4d82-b9c1-23f8b3bebdd2.mp4

Anyways, we will have enough time to iterate so we don't need to have it perfect now.

**Comment by elliottprogrammer** [comment]
> Now, everything looks fine in my opinion. I only caught a very minor issue on mobile which you can see in the following recording.

Ahh yes, thank you. A quick fix adjusting the card flexbox on mobile. 👍  The issue shown in the screen recording should be resolved now. 🙂

**Comment by elliottprogrammer** [comment]
And yes, I tend to agree also that the scrollbar on the dropdown is probably the way to go. Thanks for your input! 🙂

File: client/my-sites/checkout/checkout-thank-you/licensing-thank-you-auto-activation.tsx [hunk 6]
```diff
@@ -0,0 +1,240 @@
...
+			onClick: () => {
+				setSelectedSite( 'activate-license-manually' );
+			},
+		},
+	};
+	const selectDropdownItems = siteSelectOptions && [
+		...siteSelectOptions,
+		...[ lastSelectOption ],
+	];
+	const selectedItem = find( selectDropdownItems, 'props.selected' );
```
Comment (rcanepa) [inline client/my-sites/checkout/checkout-thank-you/licensing-thank-you-auto-activation.tsx]: I don't really know if using `lodash` here will increase the size of the bundle, but to be on the safe side, I suggest replacing `lodash.find` with the JavaScript native version of `find`.

So,
```
const selectedItem = find( selectDropdownItems, 'props.selected' );
```

would become

```
const selectedItem = selectDropdownItems.find( ( item ) => item.props.selected );
```

**Review by rcanepa — APPROVED** [review]
LGTM!

Other than the comment about `lodash`, this looks great! We can now build the rest of the UIs using this one as a template.

Amazing work, Bryan!

File: client/my-sites/checkout/checkout-thank-you/licensing-thank-you-auto-activation.tsx (hunk 6 above)
Comment (elliottprogrammer) [reply client/my-sites/checkout/checkout-thank-you/licensing-thank-you-auto-activation.tsx]: Yes right, good call! 🙂

**Comment by a8ci18n** [comment]
This Pull Request is now available for translation here: https://translate.wordpress.com/deliverables/6940614

Thank you @elliottprogrammer for including a screenshot in the description! This is really helpful for our translators.

**Comment by a8ci18n** [comment]
Translation for this Pull Request [has now been finished](https://translate.wordpress.com/deliverables/overview/6940614).

**Pull request merged**